presto-clp/src/main/java/com/yscope/presto/ClpConnectorFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yscope.presto;
+
+import com.facebook.airlift.bootstrap.Bootstrap;
+import com.facebook.airlift.json.JsonModule;
+import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.spi.ConnectorHandleResolver;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorContext;
+import com.facebook.presto.spi.connector.ConnectorFactory;
+import com.facebook.presto.spi.function.FunctionMetadataManager;
+import com.facebook.presto.spi.function.StandardFunctionResolution;
+import com.facebook.presto.spi.relation.RowExpressionService;
+import com.google.inject.Injector;
+
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class ClpConnectorFactory
+        implements ConnectorFactory
+{
+    @Override
+    public String getName()
+    {
+        return "clp";
+    }
+
+    @Override
+    public ConnectorHandleResolver getHandleResolver()
+    {
+        return new ClpHandleResolver();
+    }
+
+    @Override
+    public Connector create(String catalogName, Map<String, String> config, ConnectorContext context)
+    {
+        requireNonNull(catalogName, "catalogName is null");
+        requireNonNull(config, "config is null");
+        try {
+            Bootstrap app = new Bootstrap(new JsonModule(), new ClpModule(), binder -> {
+                binder.bind(TypeManager.class).toInstance(context.getTypeManager());
+                binder.bind(NodeManager.class).toInstance(context.getNodeManager());
+                binder.bind(FunctionMetadataManager.class).toInstance(context.getFunctionMetadataManager());
+                binder.bind(StandardFunctionResolution.class).toInstance(context.getStandardFunctionResolution());
+                binder.bind(RowExpressionService.class).toInstance(context.getRowExpressionService());
+            });
```

Comment:
I saw there is a ClpModule class, shall we add it here, because there is a place to register the module. Also, otherwise the catalogName this parameter is never used
```suggestion
            },
            locationModule,
            new ClpModule(catalogName));
```

Reply:
We've already registered ClpModule, probably use the catalog name as the connector id

Reply:
ClpModule is already added